python/cudf_polars/tests/test_join.py

```diff
@@ -85,16 +86,32 @@ def test_coalesce_join(left, right, how, nulls_equal, join_expr):
     query = left.join(
         right, on=join_expr, how=how, nulls_equal=nulls_equal, coalesce=True
     )
-    assert_gpu_result_equal(query, check_row_order=how == "left")
+    assert_gpu_result_equal(query, check_row_order=False)
 
 
 def test_left_join_with_slice(left, right, nulls_equal, zlice):
     q = left.join(right, on="a", how="left", nulls_equal=nulls_equal, coalesce=True)
 
     if zlice is not None:
+        # neither polars nor cudf guarantee row order.
+        # left.join(right).slice(slice) is a fundamentally sensitive to
+        # the row ordering of the join algorithm. So we can just check
+        # the things that invariant to the ordering.
         q = q.slice(*zlice)
 
-    assert_gpu_result_equal(q)
+        engine = get_default_engine()
+
+        # Check the number of rows
+        assert_gpu_result_equal(q.select(pl.len()))
+
+        # Check that the schema matches
+        result = q.collect(engine=engine)
```

Comment:
```suggestion
        # Check the number of rows
        assert_gpu_result_equal(q.select(pl.len()))

        # Check that the schema matches
        result = q.collect(engine=get_default_engine())
```

Reply:
Can we do `assert_gpu_result_equal(..., check_order=False)`?

Reply:
According to https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.join.html, the fact that it happens to pass now is just a coincidence of both polars and cudf-polars' single-partition streaming executor preserving the row order:

> `maintain_order{‘none’, ‘left’, ‘right’, ‘left_right’, ‘right_left’}`
> Which DataFrame row order to preserve, if any. Do not rely on any observed ordering without explicitly setting this parameter, as your code may break in a future release.

So I can scope this change to just the case that currently fails (multi-partition streaming executor) if we want, if we're willing to tolerate future changes to polars / cudf-polars potentially causing this test to fail.

Reply:
Oh right, I hadn't looked at my `maintain_order` PR in a while. I'm fine with this test.